ui/intermittent-failures/ErrorBoundary.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { hasError } from './redux/actions';
+import { prettyErrorMessages } from './constants';
+
+class ErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  componentDidCatch() {
+    const { errorFound, stateName } = this.props;
+
+    // display fallback UI and reset isFetching to turn off the loading spinner
+    this.setState({ hasError: true }, () => errorFound(stateName));
+
+    // TODO: set up a logger to record error and { componentStack }
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return <p className="text-danger py-2">{prettyErrorMessages.default}</p>;
+    }
+    return this.props.children;
+  }
+}
+
+ErrorBoundary.propTypes = {
+  stateName: PropTypes.string.isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.shape({}),
+    PropTypes.bool,
+  ]),
+  errorFound: PropTypes.func,
+};
+
+ErrorBoundary.defaultProps = {
+  errorFound: null,
```

Comment:
This looks like it would throw an exception within the ``ErrorBoundary`` if we fall back to the default.  Seems like ``errorFound`` should be ``isRequired`` or else default to a dummy function like ``() => {}``.  Otherwise, it will die on line 18.

Reply:
Ah, good catch - I'll fix that.

Reply:
I just realized I did that because the errorFound prop is supplied from the Redux mapDispatchToProps on line 45, but I marked it as required regardless :)